### DISCUSSION TITLE
一个关于 默认播放器模式 的问题

### DISCUSSION THREAD

**Discussion by Dlineou** [the1812/Bilibili-Evolved#2815] [提问 / Q&A] [answered]
只安装默认播放器一个插件的情况下，如果用鼠标中键在后台打开网页然后在点过去，那么默认播放器的样式就没法起效必须，刷新一下才能生效，而鼠标左键直接打开网页就能生效，V1版本就没这个毛病，哪位大佬能解答一下是哪里的问题，或者定位BUG修复下？

**Comment by FoundTheWOUT** [comment, marked as answer]
哇，真的细，我一直觉得哪里有点怪，原来是这个点。确实挺奇怪的...
我再仔细测试一下，用中键打开网页也是可以**生效**的，只要手速够快（中键完立刻切过去）。
是有点边缘情况在里面

**Reply by Dlineou** [reply]
也许是加载时间的问题，不知道怎么提交BUG，老哥能帮忙提交一下吗

**Reply by FoundTheWOUT** [reply]
没事，不用开 issues 了，我马上修改。我用默认播放器模式也挺多的

**Reply by Dlineou** [reply]
老大啥时候能上线啊？跟着2.0.9吗

**Reply by FoundTheWOUT** [reply]
~~2.0.9其实已经发版了，1812忘记了改版本~~
2.0.10应该可以上

**Reply by the1812** [reply]
我后来改了的(

**Reply by Dlineou** [reply]
大佬，我测试了最新的2.0.10测试版，这个问题没完全修好啊，用中键打开的网页只有马上切换过去默认宽屏才能生效，如果把网页打开好几个放在后台一个个看，默认宽屏就不能生效了。测试的浏览器是WIN11下的EDGE浏览器。

**Reply by FoundTheWOUT** [reply]
看看你的默认播放器模式版本是多少，你试试单独更新一下`默认播放器模式`

**Reply by Dlineou** [reply]
哔哩哔哩播放器版本2.73.1-07859f81，默认播放器模式版本93accf2ac应该也是最新了，更新后没变化，依然还是中键放在后台的视频，再打开不会自动应用格式，必须刷新才行。

**Reply by the1812** [reply]
最新的是 `b6ea3caa7`
你可以用链接覆盖安装下看看: https://raw.githubusercontent.com/the1812/Bilibili-Evolved/preview/registry/dist/components/video/player/default-mode.js

**Reply by Dlineou** [reply]
安装完这个版本默认宽屏完全用不了了。。。现在不止中键打开的网页不生效，左键打开的也不能生效，甚至网页刷新了都不生效了。。。

**Reply by the1812** [reply]
@FoundTheWOUT 那个 loadButton 失败了
```
Uncaught (in promise) Error: lazy player settings panel failed! panelSelector = .bilibili-player-video-btn-setting-wrap
    at loadLazyPlayerSettingsPanel (webpack://@bevo/core/./src/core/utils/lazy-panel.ts?:59)
    at async i (eval at <anonymous> (webpack://@bevo/core/./src/core/external-input.ts?), <anonymous>:1:943)
    at async eval (eval at <anonymous> (webpack://@bevo/core/./src/core/external-input.ts?), <anonymous>:1:1198)
```

**Reply by FoundTheWOUT** [reply]
挺奇怪的，我这边（2.73.1）没有测出问题

**Reply by the1812** [reply]
怀疑是 loadLazyPlayerSettingsPanel 的时机太早了, 那个设置图标的 mouseover 事件都还没绑上去

**Reply by FoundTheWOUT** [reply]
`loadButton` 定义的时候 `loadLazyPlayerSettingsPanel` 不会执行吧？
真正执行 `loadLazyPlayerSettingsPanel` 的时候*感觉*上已经等够时间了，又是等待playerReady，又是等待视频元素的...
另外一点，我在其它（需要等待视频的）地方，比如自动关灯，都是用`videoChange`这个api做的，没遇到页面挂后台功能就执行不了的Bug，是不是可以考虑一下这个api

**Reply by the1812** [reply]
可以试试 `videoChange`, 这种神秘 bug 实在搞不懂啥原因(

**Reply by FoundTheWOUT** [reply]
emmm，试了一圈又感觉没啥问题，而且想了想，这里不需要用`videoChange`，因为切P的时候实际不需要重新挂载什么东西，如果用了反而会导致其它问题（比如之前说的蹦回去）。
@Dlineou 你试试用用其它模式正常不，比如`网页全屏`。

**Reply by Dlineou** [reply]
> emmm，试了一圈又感觉没啥问题，而且想了想，这里不需要用，因为切P的时候实际不需要重新挂载什么东西，如果用了反而会导致其它问题（比如之前说的蹦回去）。 @Dlineou你试试用用其它模式正常不，比如。`videoChange``网页全屏`

不行啊，啥模式都不生效，而且重新安装也不能退回93accf2ac这个版本了。

**Reply by FoundTheWOUT** [reply]
那就真的太奇怪了，你看看控制台有没有报错信息，是不是和1812的一样。
而且，`全屏`模式的逻辑和 `宽屏`、`网页全屏`是不一样的，如果`全屏`都有问题那就真的奇怪了

**Reply by Dlineou** [reply]
> 那就真的太奇怪了，你看看控制台有没有报错信息，是不是和1812的一样。 而且，`全屏`模式的逻辑和 `宽屏`、`网页全屏`是不一样的，如果`全屏`都有问题那就真的奇怪了

测试了一下，又出现了更诡异的BUG，好消息是全屏能触发，坏消息是选择的是全屏，但触发的是网页全屏，而且按右下角按钮是退不出去网页全屏模式的，必须按ESC才能退出，即使刷新也会重复触发这个BUG。控制台有64个报错，但我不知道怎么发给你。

**Reply by FoundTheWOUT** [reply]
![image](https://user-images.githubusercontent.com/47922588/149624531-4db00b40-6b9f-442a-841c-f3de1874df2b.png)
关于全屏触发成网页全屏，也许不是bug，是feature，不知道是不是没办法全屏
anyway，你按照上面的图把报错贴一下（记得先切回宽屏）

**Reply by Dlineou** [reply]
> ![image](https://user-images.githubusercontent.com/47922588/149624531-4db00b40-6b9f-442a-841c-f3de1874df2b.png) 关于全屏触发成网页全屏，也许不是bug，是feature，不知道是不是没办法全屏 anyway，你按照上面的图把报错贴一下（记得先切回宽屏）

你看看这个对吗？![image](https://user-images.githubusercontent.com/62940580/149624859-806277f4-007e-4345-b40b-40b5f1564a25.png)

**Reply by FoundTheWOUT** [reply]
你是用什么浏览器

**Reply by Dlineou** [reply]
> 你是用什么浏览器

![image](https://user-images.githubusercontent.com/62940580/149624929-52f825e0-2ace-4ac1-a901-74bd43394f48.png)

**Reply by Dlineou** [reply]
> 你是用什么浏览器

另外还有一个刚发现的问题，就是视频播放页这个地方会错位，但是首页没任何问题。
![image](https://user-images.githubusercontent.com/62940580/149625332-702b6347-f81c-41b8-9e6a-92582a779ac4.png)
这个是首页没错位的样子
![image](https://user-images.githubusercontent.com/62940580/149625386-5dca9de0-e960-4d0b-9f2f-96505f89afb5.png)

**Reply by FoundTheWOUT** [reply]
#2740
更新一下自定义顶栏组件

**Reply by Dlineou** [reply]
更新了最新的预览版本，确实解决了中键后台打开视频的问题，但貌似视频的位置又出现了新的问题。

这个是老版自动宽屏之后的效果（中键打开网页不能应用宽屏的版本）
![2](https://user-images.githubusercontent.com/62940580/149955994-c748b13b-4aa2-45d2-ab8c-1912d4d87896.png)

这个是最新预览版的效果
![3](https://user-images.githubusercontent.com/62940580/149956065-b6436289-7ece-48a2-8d6b-ddd2f1d9463c.png)

明显可以看出新版本视频播放的位置存在问题，虽然宽屏应用了，但是位置有问题，下方的画面直接被遮挡了。希望大佬能修复一下！

**Reply by FoundTheWOUT** [reply]
#483

**Reply by Dlineou** [reply]
比较好奇的是为啥老版本的就没这个问题，除了不能后台打开的时候应用其他都挺好，新版反而出现这种定位的问题了

**Reply by the1812** [reply]
新版加入了禁用这种定位效果的功能, 也就是最终解决了 #483, 看来还是得留个开关控制啊
